Sources/AuthFoundation/User Management/Credential.swift

```diff
@@ -67,9 +67,28 @@ public class Credential {
         try coordinator.with(token: token)
     }
     
+    /// Returns the ``Credential`` that matches the given ID.
+    /// - Parameter id: ID for the credential to return.
+    /// - Returns: Credential matching the ID.
+    public static func with(id: String) throws -> Credential? {
+        try coordinator.with(id: id)
```

Comment:
I'm utilizing static dependency injection here for the purposes of testability. There's no occasion where a developer would need to override this particular mechanism, but I also try to avoid using singletons wherever possible. In this case, we can't avoid using static/singleton functions, so I use this dependency injection to be able to more reliably test both sides of the singleton.